apollo-runtime/src/main/java/com/apollographql/android/cache/normalized/GraphQLResponseNormalizer.java

```diff
@@ -0,0 +1,137 @@
+package com.apollographql.android.cache.normalized;
+
+import com.apollographql.android.api.graphql.Field;
+import com.apollographql.android.api.graphql.Operation;
+import com.apollographql.android.cache.ResponseReaderShadow;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.Stack;
+
+import javax.annotation.Nullable;
+
+public class GraphQLResponseNormalizer implements ResponseReaderShadow {
+
+  private Stack<List<String>> pathStack;
```

Comment:
Stack is based on Vector that is synchronized. Do we really need synchronization? Question about performance, should we replace with light weight impl of stack structure?